lib/api/src/identity/Identity.ts

```diff
@@ -0,0 +1,22 @@
+
+export abstract class Identity {
+
+  /**
+   * Generates a url to be used as authentication redirect url.
+   *
+   * @param path A pathname to append to the origin
+   * @return Authentication redirect url
+   */
+  public abstract getRedirectUrl (path: string) : string
+
+  /**
+   * Performs first steps of OAuth2 flow, including authenticating user with the
+   * service provider and and handling client authorization.
+   *
+   * @param url The url required by service provider to grant access token
+   * @param interactive Whether flow should complete/fail silently
```

Comment:
Possibly mention that the default is `true`.

Reply:
👍

Reply:
Actually this is a bit odd now as abstract class specifies `interactive` as optional and cannot assign a default value since abstract but now specifies 'true by default' in doc so that concrete - which implements the parameter as non-optional with default value - can inherit.... :grin: is that good practice???